subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -733,6 +736,30 @@ static int tls_opt_peer_verify_set(struct net_context *context,
 	return 0;
 }
 
+static int tls_opt_role_set(struct net_context *context, const void *optval,
+			    socklen_t optlen)
+{
+	int *role;
+
+	if (!optval) {
+		return -EFAULT;
+	}
+
+	if (optlen != sizeof(int)) {
+		return -EINVAL;
+	}
```

Comment:
why this check ? optlen is a socklen_t that is a typedef to size_t. If size_t happens to be different of int (or unsigned int) this will fail. If you really need something of int's size, shouldn't be better change this parameter's type ?